cmd/bcpd/app/app.go

```diff
@@ -44,6 +45,8 @@ func Chain(minFee x.Coin, authFn x.Authenticator) app.Decorators {
 		namecoin.NewFeeDecorator(authFn, minFee),
 		// cannot pay for fee with hashlock...
 		hashlock.NewDecorator(),
+		// make sure we execute all the transactions in batch before the save point
+		batch.NewDecorator(),
```

Comment:
req: the batch decorator should be ordered first after `utils.NewSavepoint().OnCheck()` to apply the other decorators at each message of the batch. Otherwise the context could not be set for them.